### ISSUE TITLE
Troubles with get_joy_button_string

### ISSUE THREAD

**Issue by palrogg** [Algorithmus/DiasporicCrypt#2]
I'm getting troubles launching the game from Godot. The blue intro background appears, the music plays, but the first menu doesn't appear.
I get this error in Godot 2.1:
```Invalid call. Nonexistent function 'get_joy_button_string' in base 'InputDefault'.```
Maybe you don't use the same version of Godot?

**Comment by Algorithmus** [comment]
Well, it does appear to be a function: http://docs.godotengine.org/en/stable/classes/class_input.html#class-input-get-joy-button-string
https://github.com/godotengine/godot/blob/2.1/main/input_default.h

I'm not getting that error. Can you provide more information about which file this happens in?

**Comment by palrogg** [comment]
Yes, it happens in inputkey.gd lines 37 and 45, as if my keyboard wasn’t recognized as current input:
<img width="574" alt="screenshot_get_joy" src="https://user-images.githubusercontent.com/12730304/34323033-efc0ea74-e837-11e7-8483-ac1a99f400d5.png">

**Comment by Algorithmus** [comment]
I'm still unable to reproduce this problem, so you may have to do some debugging yourself. It appears to have something to do with the keyboard/joystick detection. Try with or without a joystick and find out if Globals.get("current_input") from line 24 returns anything.

**Comment by Algorithmus** [comment]
@palrogg I think I know what the cause might be. It didn't stop the game for me, but in Godot 3, I am getting warning messages about current_input in the console because I tried to access ProjectSettings with get() instead of checking with has_setting(). This should be fixed in the Godot 3 port, which is now merged into master. If you have Godot 3, perhaps you can try it again now and see if the same problem still comes up.
